### ISSUE TITLE
Is this project a combination of edbg and free-dap?

### ISSUE THREAD

**Issue by ooxi** [ataradov/embedded-swd#1]
I already wondered in the past whether it would be possible to combine edbg and free-dap into one project in order to increase flashing speed (I guess the slowest path is the USB HID communication between edbg and free-dap). I know that Adafruit has done something similar called [Adafruit DAP](https://github.com/adafruit/Adafruit_DAP).

Therefore my question is: what's the difference between this project, free-dap, edbg and Adafruit DAP :-)

**Comment by ataradov** [comment]
This is not really a "project". There are no plans to maintain this as all. This is just a demo of how one MCU can program another MCU directly over SWD interface. I made this some time ago because someone asked for it, but I never made it public. Until a few days ago, when I got asked again and had to dig to find this code. So I published it to make it easier to find the next time.

This is exactly it - the combination of edbg + free-dap in one place.

Adafruit DAP appears to be the same concept, but wrapped differently according to their needs. This is the intended way of doing it - take the code and modify to suit your needs.

This is very convenient as a substitute for a bootlaoder with lots of added benefits over the bootloader.
